lib/chef/provider/package.rb

```diff
@@ -464,10 +464,7 @@ def current_version_array
 
       # @return [Array] new_version(s) as an array
       def new_version_array
-        @new_version_array ||=
-            [ new_resource.version ].flatten.map do |v|
-              ( v.nil? || v.empty? ) ? nil : v
-            end
+        [ new_resource.version ].flatten
```

Comment:
English is not mother tongue, I'll try to be more clear and detailed this time :).

When we're calling `new_version_array` [first time](https://github.com/ojab/chef/blob/yum_fix/lib/chef/provider/package/yum.rb#L1049) to get second arg to `parse_dependency`, we're setting `@new_version_array` to `[nil]` and it's value cached.

Then to determine which pkgs should be installed we're calling [target_version_array](https://github.com/chef/chef/blob/99f236ac1c6996a467c8f4dda256e528c4355259/lib/chef/provider/package.rb#L335) that calls [each_package](https://github.com/chef/chef/blob/99f236ac1c6996a467c8f4dda256e528c4355259/lib/chef/provider/package.rb#L436). `each_package` gets `new_version` from our cached `@new_version_array == [nil]`, so `new_version` is always nil not matter what.

So `if new_version` branch in `target_version_array` is always false and  we're not updating pkg that is already installed but doesn't match out version constrain.

Hope that I'm more clear this time.